### PR TITLE
Implement generic timm classification models in library

### LINKED ISSUE #7097: Library: Implement generic timm classification models
Consolidate `timm` classification execution in the `library` package by implementing task-generic recipes (`MULTI_CLASS_CLS` and `MULTI_LABEL_CLS`) and refining model/backbone initialization. 

`TimmBackbone` must be updated to use headless feature extraction (num_classes=0), preserving each model’s default pooling and dynamically retrieving standard input parameters (`input_size`, `mean`, `std`, `interpolation`). 

Additionally, implement dynamic, architecture-aware optimizer creation inside the `TimmModel` wrappers (mapping transformer families like vit/swin/deit to `AdamW` and others to `SGD`) using `timm.optim.create_optimizer_v2` with the configured learning rate.

See [design 3.1-3.3](https://github.com/open-edge-platform/geti/blob/develop/library/docs/design/timm-integration.md#3-required-library-changes) for more details.

### PR DESCRIPTION
**Pull request by itallix**
### Summary

- [x] New generic recipes `timm_generic.yaml` for both `MULTI_CLASS_CLS` & `MULTI_LABEL_CLS`
- [x] Added builder function for architecture-aware timm optimizer + getter for preprocessing params
- [x] `TimmBackbone.forward` overridden to return pooled global feature embeddings

### Checklist

- [x] The PR title and description are clear and descriptive
- [ ] I have manually tested the changes
- [x] All changes are covered by automated tests
- [x] All related issues are linked to this PR (if applicable)
- [ ] Documentation has been updated (if applicable)

### CODE REVIEW

File: library/src/getitune/backend/lightning/models/classification/backbones/timm.py [hunk 1]
```diff
@@ -10,10 +10,59 @@
...
+        model_name: Name of the timm model (as passed to `timm.create_model`).
+        learning_rate: Base learning rate for the optimizer.
+
+    Returns:
+        A callable (partial of `timm.optim.create_optimizer_v2`) that produces the
+        optimizer when called with the model's parameters.
+    """
+    kwargs = (
+        {"opt": "adamw", "lr": learning_rate, "weight_decay": 0.05}
+        if _is_transformer_family(model_name)
```
Comment (kprokofi) [inline library/src/getitune/backend/lightning/models/classification/backbones/timm.py]: In the generated json snapshot we already have this info as "AdamW" optimizer. Can we just reuse it?
Do you think is it OK to use "weight_decay": 0.05 for all models?

File: library/src/getitune/backend/lightning/models/classification/backbones/timm.py [hunk 2]
```diff
@@ -10,10 +10,59 @@
...
+_TRANSFORMER_PREFIXES = ("vit", "deit", "beit", "swin", "cait", "xcit", "maxvit", "coat", "twins", "pvt")
+
+
+def _is_transformer_family(model_name: str) -> bool:
+    """Check whether a timm model name belongs to a transformer-like family."""
+    family = model_name.split("_")[0].split(".")[0].lower()
+    return family.startswith(_TRANSFORMER_PREFIXES)
+
+
+def build_timm_optimizer_fn(model_name: str, learning_rate: float) -> Callable[..., torch.optim.Optimizer]:
```
Comment (kprokofi) [inline library/src/getitune/backend/lightning/models/classification/backbones/timm.py]: Can we move these functions  ( + get_preprocessing_params) to utils?

File: library/src/getitune/backend/lightning/models/classification/multilabel_models/timm_model.py [hunk 3]
```diff
@@ -54,16 +59,22 @@ class TimmModelMultilabelCls(TimmWeightsLoader, LightningMultilabelClsModel):
     def __init__(
         self,
         label_info: LabelInfoTypes,
+        learning_rate: float | None = None,
```
Comment (kprokofi) [inline library/src/getitune/backend/lightning/models/classification/multilabel_models/timm_model.py]: Instead of introducing new learning rate parameter and None for optimizer, we could create a factory TimmOptimizer that will choose a base optimizer and default LR if not set in cfg. 

In timm_generic.ymal we will define:

```
class_path: getitune.backend.lightning.models.classification.multiclass_models.timm_model.TimmModelMulticlassCls
  init_args:
    label_info: 1000
    model_name: DYNAMIC
    optimizer: getitune.backend.lightning.models.classification.multiclass_models.timm_model.optimizer.TimmOptimizer
        lr: DYNAMIC
```

File: library/src/getitune/backend/lightning/models/classification/backbones/timm.py (hunk 1 above)
Comment (itallix) [reply library/src/getitune/backend/lightning/models/classification/backbones/timm.py]: It should be fine to start with 0.05, though typically it is between 0.01 and 0.05. We can also refine it later based on a specific family/models.

File: library/src/getitune/backend/lightning/models/classification/backbones/timm.py (hunk 2 above)
Comment (itallix) [reply library/src/getitune/backend/lightning/models/classification/backbones/timm.py]: Moved

File: library/src/getitune/backend/lightning/models/classification/multilabel_models/timm_model.py (hunk 3 above)
Comment (itallix) [reply library/src/getitune/backend/lightning/models/classification/multilabel_models/timm_model.py]: Added `getitune.backend.lightning.models.classification.utils.timm.TimmOptimizer`

**Review by kprokofi — APPROVED** [review]
minor comment to not stop your work:
we could move optimizer implementation to the "optimizer/" folder in the classification root

Image

Something like optimizer/timm.py or place in the loss folder. IMO, it would look cleaner than in utils.

Other than that, LGTM
